examples/custom_plot_manipulation/src/main.rs

```diff
@@ -0,0 +1,128 @@
+//! This example shows how to implement custom gestures to pan and zoom in the plot
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use eframe::egui::{self, plot::Legend, DragValue, Event};
+use egui::{
+    plot::{Line, PlotPoints},
+    Vec2,
+};
+
+fn main() -> Result<(), eframe::Error> {
+    let options = eframe::NativeOptions::default();
+    eframe::run_native(
+        "Plot",
+        options,
+        Box::new(|_cc| Box::<PlotExample>::default()),
+    )
+}
+
+struct PlotExample {
+    lock_x: bool,
+    lock_y: bool,
+    ctrl_to_zoom: bool,
+    shift_to_horizontal: bool,
+    zoom_speed: f32,
+    scroll_speed: f32,
+}
+
+impl Default for PlotExample {
+    fn default() -> Self {
+        Self {
+            lock_x: false,
+            lock_y: false,
+            ctrl_to_zoom: false,
+            shift_to_horizontal: false,
+            zoom_speed: 1.0,
+            scroll_speed: 1.0,
+        }
+    }
+}
+
+impl eframe::App for PlotExample {
```

Comment:
I tried running this example, but I didn't understand what it was supposed to do. Please make it self-descriptive!

Reply:
I added some explanations in the UI 

![image](https://github.com/emilk/egui/assets/43298013/5689e6a7-5f82-4a70-aa18-33f316217d9e)